### PR TITLE
Lastseen relative

### PR DESCRIPTION
**Pull request by nelsnelson**
> Have it compute the elapsed relative time as well, e.g. `playername was last seen Wed 8, May, 8:24pm (6 days, 12 hours, 34 minutes ago)`.
-- @totemo, https://nerdnu.slack.com/archives/GJ4MJQRT8/p1557832333087300

### CODE REVIEW

**Comment by nelsnelson** [comment]
Should probably not merge this until after https://github.com/bermudalocket/LastSeen/pull/1 is merged and this branch is rebased with the new master.

Edit: This branch has been rebased with master after #1 was merged.  This PR is now ready to be merged when accepted.

File: src/main/java/com/bermudalocket/lastseen/LastSeen.java [hunk 1]
```diff
@@ -126,6 +164,19 @@ private static String longToDate(Long time) {
...
+    /**
+     * Turns the given timestamp into a string describing the relative date
+     * in English to the current time now.
+     *
+     * @param time the timestamp.
+     * @return a string describing the relative date.
+     */
+    private static String longToRelativeDate(Long time) {
+        CALENDAR.setTimeInMillis(time);
+        return PRETTY_TIME_FORMAT.format(CALENDAR.getTime());
```
Comment (bermudalocket) [inline src/main/java/com/bermudalocket/lastseen/LastSeen.java]: Can you elaborate more on this change?

File: pom.xml [hunk 2]
```diff
@@ -41,6 +50,27 @@
                 <configuration>
                     <source>1.8</source>
                     <target>1.8</target>
+                    <compilerArgument>-Xlint:unchecked</compilerArgument>
```
Comment (bermudalocket) [inline pom.xml:53]: Why is this necessary?

File: pom.xml [hunk 3]
```diff
@@ -41,6 +50,27 @@
...
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-assembly-plugin</artifactId>
```
Comment (bermudalocket) [inline pom.xml:58]: What is the advantage of the Maven assembly plugin over the Maven shade plugin?

File: src/main/java/com/bermudalocket/lastseen/LastSeen.java (hunk 1 above)
Comment (nelsnelson) [reply src/main/java/com/bermudalocket/lastseen/LastSeen.java]: Sorry, I neglected to include a description.  I probably should have created an issue to track.  Will do so in the future.  See new PR description: https://github.com/bermudalocket/LastSeen/pull/2#issue-279588324

File: pom.xml (hunk 2 above)
Comment (nelsnelson) [reply pom.xml:53]: It helps to know where unchecked assignments are happening in code as a matter of best practice code smell reduction.

File: pom.xml (hunk 3 above)
Comment (nelsnelson) [reply pom.xml:58]: The maven assembly plugin supports the compilation of a fat jar which includes the exploded and manifested inclusion of all dependencies in the distributable artifact.  This enables simple distribution and installation of the plugin to the server.

Does the maven shade plugin do something similarly?  What are the advantages of the shade plugin?

File: pom.xml (hunk 2 above)
Comment (bermudalocket) [reply pom.xml:53]: Ah, what IDE are you using (if at all)? IntelliJ can detect unchecked assignments; I believe it's one of the checks enabled by default. Either way, good to know. I had not seen the `Xlint` flag before.

File: pom.xml (hunk 3 above)
Comment (bermudalocket) [reply pom.xml:58]: The shade plugin is a more robust compared to the assembly plugin (see examples [here](http://maven.apache.org/plugins/maven-shade-plugin/) and [here](http://maven.apache.org/plugins/maven-assembly-plugin/examples/index.html)) though it probably doesn't make a difference here.

Comment (nelsnelson) [reply pom.xml:58]: Oh!  Good to know!  I'll read the documents at those links.  Do they include any examples/tutorials on usage?

File: pom.xml (hunk 2 above)
Comment (nelsnelson) [reply pom.xml:53]: I use Eclipse, or just Sublime Text, depending on what I'm trying to accomplish.
